akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/ReplayingEvents.scala

```diff
@@ -116,10 +116,8 @@ private[akka] final class ReplayingEvents[C, E, S](
 
             def handleEvent(event: E): Unit = {
               eventForErrorReporting = OptionVal.Some(event)
-              state = state.copy(
-                seqNr = repr.sequenceNr,
-                state = setup.eventHandler(state.state, event),
-                eventSeenInInterval = true)
+              state = state.copy(seqNr = repr.sequenceNr)
+              state = state.copy(state = setup.eventHandler(state.state, event), eventSeenInInterval = true)
```

Comment:
I don't get why we need two passes here. Is that a leftover from some other experiment? If not, can we add a comment explaining the reason for that.

Reply:
ok, I get it. We need to be able to access it using `EventSourcedBehavior.lastSequenceNumber(ctx)}`

Reply:
yes